collectors/service.go

```diff
@@ -42,6 +42,12 @@ var (
 		[]string{"namespace", "service"}, nil,
 	)
 
+	descServiceSpecServiceType = prometheus.NewDesc(
+		"kube_service_spec_type",
+		"Type about service.",
+		[]string{"namespace", "service", "clusterIP", "type"}, nil,
```

Comment:
If I understand correctly `clusterIP` label is not used to uniquely identify this metric so it would rather belong in the info metric. Aside from that, the naming convention would be `cluster_ip`.